geps/gep-1713/index.md

```diff
@@ -0,0 +1,408 @@
+
+	// Name is the name of the referent.
+	Name ObjectName `json:"name"`
+
+	// Namespace is the namespace of the referenced object. When unspecified, the local
+	// namespace is inferred.
+	//
+	// Support: Core
+	//
+	// +optional
+	Namespace *Namespace `json:"namespace,omitempty"`
+}
+```
+
+#### YAML
+
+Below shows an example of an end to end configuration with Gateway merging.
+Here we define a parent resource, which allows children from the same namespace.
+A single listener is defined in the parent.
+
+The child Gateway attaches to this Gateway and specifies an additional listener.
+
+```yaml
+apiVersion: gateway.networking.k8s.io/v1beta1
+kind: Gateway
+metadata:
+  name: parent-gateway
+spec:
+  gatewayClassName: example
+  infrastructure:
+    allowedChildren:
+      namespaces:
+        from: Same
+  listeners:
+  - name: common-monitoring
+    port: 8080
+    protocol: HTTP
+---
+apiVersion: gateway.networking.k8s.io/v1beta1
+kind: Gateway
+metadata:
+  name: child-gateway
+spec:
+  gatewayClassName: example
+  infrastructure:
+    attachTo:
+      name: parent-gateway
+      kind: Gateway
+      group: gateway.networking.k8s.io
+  listeners:
+  - name: domain-a
+    hostname: a.example.com
+    protocol: HTTP
+    port: 80
+```
+
+Logically, this is equivalent to a single Gateway as below:
+
+```yaml
+apiVersion: gateway.networking.k8s.io/v1beta1
+kind: Gateway
+metadata:
+  name: gateway
+spec:
+  gatewayClassName: example
+  listeners:
+  - name: common-monitoring
+    port: 8080
+    protocol: HTTP
+  - name: domain-a
+    hostname: a.example.com
+    protocol: HTTP
+    port: 80
+```
+
+### Semantics
+
+#### Route Attaching
+
+Routes MUST be able to specify a child Gateway as a `parentRef` and make use of the fields in `ParentReference` to help target a specific listener.
+If no listener is targeted (`sectionName`/`port` are unset) then the Route references all the listeners on the child Gateway. It MUST NOT attach
+to a listener on a parent or sibling Gateway.
+
+```yaml
+apiVersion: gateway.networking.k8s.io/v1beta1
+kind: HTTPRoute
+metadata:
+  name: httproute-example
+spec:
+  parentRefs:
+  - name: child-gateway
+    sectionName: metrics
+```
+
+Routes can only bind to listeners *directly* defined in the `Gateway` referenced.
+For instance, the following route referencing a listener defined in the parent `Gateway`, but attaching to the child `Gateway` is not valid.
+This will be reported in [status](#status-fields).
+
+```yaml
+apiVersion: gateway.networking.k8s.io/v1beta1
+kind: HTTPRoute
+metadata:
+  name: httproute-example
+spec:
+  parentRefs:
+  - name: child-gateway
+    sectionName: section-in-parent
+```
+
+#### Merging Spec Fields
+
+##### Validation
+
+A parent and child MUST have the same `gatewayClassName`.
+This will be detected by the implementation and reported in [status](#status-fields).
+
+A child resource MUST not set any `spec.infrastructure` fields beyond `attachTo`, and cannot set `spec.address`.
```

Comment:
Would you consider a case where there can be **some** listeners in a Gateway that are never supposed to merge, and those would need to maintain a `spec.address` in that case.

Reply:
> Would you consider a case where there can be some listeners in a Gateway that are never supposed to merge, and those would need to maintain a spec.address in that case.

If you don't want the Gateways to merge then I wouldn't be use the Gateway merging feature.

>  those would need to maintain a spec.address in that case.

I'm assuming you mean different `spec.address` - in this scenario just use two different disjoint Gateways